lesson_21/app.js

```diff
@@ -0,0 +1,89 @@
+// 1. Decorator 
+
+function sumF (num){
+    return this.num + num;
+  };
+
+let cache = {};
+
+function cachesDecorator (func){
+    return function () {
+        return  cache[func.name] =  func.call(this, ...arguments); // .name - returns name of function
```

Comment:
your result will change depending on arguments you pass
so it's incorrect cache